### DISCUSSION TITLE
thoughts about the planned Multipart support?

### DISCUSSION THREAD

**Discussion by AnyChicTop** [miguelgrinberg/microdot#254] [General]
Hi Miguel,

I know it's on the roadmap. I don't know if you've started anything yet but I'd be interesting in your thoughts on how you are planning on tackling this.

- Will you be using a sliding split buffer to control memory consumption? 
- I imagine (due to the constrained device support) that temporary files are out. True?
- What about timeouts during long upload periods (due to size or poor connection)?
- Will interrupted uploads be automatically removed? Or will they be resumable?
- How about request size limiting to prevent DoS (or running out of storage)?

Thank you for Microdot. Other frameworks either included too much or were too complex for my simple use cases. Multipart is the last thing we need to have all of the basics covered.

**Comment by miguelgrinberg** [comment]
Haven't thought about it yet in detail. At a very high level my intention is to provide functions to read the parts from the stream in sequence, with some helpers to handle non-file parts in a way that is closer to how regular forms are handled. I would prefer not to have to parse the entire body and create temp files, which may not even be possible on some devices. It would be up to the application to decide how much of the stream is read with each chunk.
